pkg/dockerregistry/client.go

```diff
@@ -458,3 +700,35 @@ func unmarshalDockerImage(body []byte) (*docker.Image, error) {
 		Size:            imagePre012.Size,
 	}, nil
 }
+
+type v2layer struct {
```

Comment:
We have these types already - see https://github.com/openshift/origin/blob/master/pkg/image/api/dockertypes.go#L57-L94